numba/cuda/cudaimpl.py

```diff
@@ -523,6 +523,92 @@ def ptx_hfma(context, builder, sig, args):
     asm = ir.InlineAsm(fnty, "fma.rn.f16 $0,$1,$2,$3;", "=h,h,h,h")
     return builder.call(asm, args)
 
+
+def lower_fp16_comparison(fn, op):
+    @lower(fn, types.float16, types.float16)
+    def ptx_fp16_comparison(context, builder, sig, args):
+        temp_reg = '__$$temp3'
```

Comment:
Is this hard coding a register name? Does nvcc use `__$$temp3` every time?

Reply:
It looks like it's declaring a register with predicate type named `__$$temp3`.

Reply:
The scoping of the predicate (in the inline assembly on line 534) resolves this.